packages/muelu/test/meshtying/myXML.xml

```diff
@@ -0,0 +1,227 @@
+      <Parameter name="Strided block id" type="int" value="-1"/>
+    </ParameterList>
+
+    <ParameterList name="myAggFact1">
+      <Parameter name="factory" type="string" value="UncoupledAggregationFactory"/>
+      <Parameter name="aggregation: min agg size" type="int" value="5"/>
+      <Parameter name="aggregation: max selected neighbors" type="int" value="1"/>
+    </ParameterList>
+
+    <ParameterList name="myTentativePFact1">
+      <Parameter name="factory" type="string" value="TentativePFactory"/>
+      <Parameter name="A" type="string" value="mySubBlockAFactory1"/>
+      <Parameter name="Aggregates" type="string" value="myAggFact1"/>
+      <Parameter name="CoarseMap" type="string" value="myCoarseMap1"/>
+    </ParameterList>
+
+    <ParameterList name="myRFact1">
+      <Parameter name="factory" type="string" value="TransPFactory"/>
+      <Parameter name="P" type="string" value="myTentativePFact1"/>
+    </ParameterList>
+
+    <!-- We have to use Nullspace1 here. If "Nullspace1" is not set the
+         Factory creates the default null space containing of constant
+         vectors -->
+    <ParameterList name="myNspFact1">
+      <Parameter name="factory" type="string" value="NullspaceFactory"/>
+      <Parameter name="Fine level nullspace" type="string" value="Nullspace1"/>
+      <Parameter name="Nullspace1" type="string" value="myTentativePFact1"/>
+    </ParameterList>
+
+    <!-- BLOCK 2 (for submatrix A_{11}) -->
+    <ParameterList name="mySubBlockAFactory2">
+      <Parameter name="factory" type="string" value="SubBlockAFactory"/>
+      <Parameter name="block row"                 type="int"     value="1"/>
+      <Parameter name="block col"                 type="int"     value="1"/>
+      <Parameter name="Range map: Striding info"  type="string"  value="{ 1 }"/>
+      <Parameter name="Domain map: Striding info" type="string"  value="{ 1 }"/>
+    </ParameterList>
+
+    <!-- tell the tenative prolongator that we have 1 DOF per node on the coarse levels -->
+    <ParameterList name="myCoarseMap2">
+      <Parameter name="factory" type="string" value="CoarseMapFactory"/>
+      <Parameter name="Striding info" type="string" value="{ 3 }"/>
+      <Parameter name="Strided block id" type="int" value="-1"/>
+    </ParameterList>
+
+    <ParameterList name="mySpecialMappingFact">
+      <Parameter name="factory"                             type="string" value="InterfaceMappingTransferFactory"/>
+    </ParameterList>
+
+    <ParameterList name="myInterfaceAggs2">
+      <Parameter name="factory" type="string" value="InterfaceAggregationFactory"/>
+      <Parameter name="Aggregates" type="string" value="myAggFact1"/> <!-- use aggregates from primal block! -->
+      <Parameter name="number of DOFs per dual node" type="int" value="1"/>
+      <Parameter name="Interface Mapping Transfer Factory" type="string" value="mySpecialMappingFact"/>
+    </ParameterList>
+
```

Comment:
In order to deal with such a cyclic variable dependency between two factories, you can extend the factory definition in the xml file after the factories have been defined.

Add a new factory block here, that looks like
>    < ParameterList name="mySpecialMappingFactDeps" >
>      < Parameter name="dependency for" type="string" value="mySpecialMappingFact" / >
>      < Parameter name="CoarseDualNodeID2PrimalNodeID" type="string" value="myInterfaceAggs2" / >
>    < / ParameterList>

Reply:
The name "mySpecialMappingFactDeps" has no meaning. You need a new and unique block name for each xml block. But this name gives you a hint that you want to extend the definition of the mySpecialMappingFact block from above. The parameter "dependency for" is the keyword. The value contains the "factory name" that you want to extend (in our case "mySpecialMappingFact"). And then you just add the missing parameter dependencies. In this case you declare the myInterfaceAggs2 factory to be the generating factory for the "Coarse..." variable. You couldn't do it in the block "mySpecialMappingFact" above as the factory name "myInterfaceAggs2" was not defined, yet. But now you can do it after the "myInterfaceAggs2" block is defined.

Reply:
Please also have a look at
https://github.com/trilinos/Trilinos/blob/master/packages/muelu/src/Interface/MueLu_ParameterListInterpreter_def.hpp
particularly line 2115 ff.
There you find a more detailed description of how to handle cyclic problems. I guess, if you grep through the examples looking for "dependency for" you might (hopefully) find some examples, too.